salt/modules/yumpkg.py

```diff
@@ -3219,3 +3219,75 @@ def list_installed_patches():
         salt '*' pkg.list_installed_patches
     '''
     return _get_patches(installed_only=True)
+
+
+@salt.utils.decorators.path.which('yum-complete-transaction')
+def complete_transaction(cleanup_only=False, recursive=False, max_attempts=3):
+    '''
+    .. versionadded:: Fluorine
+
+    Execute ``yum-complete-transaction``, which is provided by the ``yum-utils`` package.
+
+    cleanup_only
+        Specify if the ``--cleanup-only`` option should be supplied.
+
+    recursive
+        Specify if ``yum-complete-transaction`` should be called recursively
+        (it only completes one transaction at a time).
+
+    max_attempts
+        If ``recursive`` is ``True``, the maximum times ``yum-complete-transaction`` should be called.
+
+    .. note::
+
+        Recursive calls will stop once ``No unfinished transactions left.`` is in the returned output.
+
+    .. note::
+
+        ``yum-utils`` will already be installed on the minion if the package
+        was installed from the Fedora / EPEL repositories.
+
+    CLI example:
+
+    .. code-block:: bash
+
+        salt '*' pkg.complete_transaction
+        salt '*' pkg.complete_transaction cleanup_only=True
+        salt '*' pkg.complete_transaction recursive=True max_attempts=5
+    '''
+
+    return _complete_transaction(cleanup_only, recursive, max_attempts, 1, [])
+
+
+def _complete_transaction(cleanup_only, recursive, max_attempts, run_count, cmd_ret_list):
+    '''
+    .. versionadded:: Fluorine
+
+    Called from ``complete_transaction`` to protect the arguments
+    used for tail recursion, ``run_count`` and ``cmd_ret_list``.
+    '''
+
+    cmd = ['yum-complete-transaction']
+    if cleanup_only:
+        cmd.append('--cleanup-only')
+
+    cmd_ret_list.append(__salt__['cmd.run_all'](
+        cmd,
+        output_loglevel='trace',
+        python_shell=False
+    ))
+
+    if (cmd_ret_list[-1]['retcode'] == 0 and
```

Comment:
Either do it implicit (not good, BTW):
```python
if (not cmd_ret_list[-1]['retcode'] and ...
```
...or explicit:
```python
if (cmd_ret_list[-1]['retcode'] == salt.defaults.exitcodes.EX_OK and ...
```

But please do not hard-code things just like that.

Reply:
Was unaware of these variables, thank you!